onnx/reference/ops/op_split_to_sequence.py

```diff
@@ -1,19 +1,41 @@
 # SPDX-License-Identifier: Apache-2.0
 # pylint: disable=C0200,W0221
 
-from .op_split import CommonSplit
+from onnx.reference.op_run import OpRun
 
 
-class SplitToSequence(CommonSplit):
+class SplitToSequence(OpRun):
+    def common_run(self, mat, split, axis, length):  # type: ignore
+        if split is None:
+            split = [1 for _ in range(mat.shape[axis])]
+        elif len(split.shape) == 0:
+            # A scaler
+            dim = mat.shape[axis]
+            length = int(split)
+            n = dim // int(length)
+            split = [length] * n
+            left = dim - length * n
+            if left > 0:
+                split.append(left)
+
+        sli = [slice(0, s) for s in mat.shape]
+        res = []
+        pos = 0
+        for spl in split:
+            sli[axis] = slice(pos, pos + spl)  # type: ignore
+            pos += spl
+            res.append(mat[tuple(sli)])
+        return tuple(res)
+
     def _run(self, mat, split=None, axis=None, keepdims=None):  # type: ignore
         if split is None:
-            num_outputs = mat.shape[axis]
+            length = None
         else:
             if len(split.shape) == 0:
-                num_outputs = split
+                length = split
             else:
-                num_outputs = split.shape[0]
-        res = list(self.common_run(mat, split, axis=axis, num_outputs=num_outputs))
+                length = split.shape[0]
+        res = list(self.common_run(mat, split, axis=axis, length=length))
```

Comment:
The line below looks incorrect. May be it is supposed to be `if keepdims == 0 and split is None` ?

Reply:
Thanks for addressing the other issues ... I think this one is still an issue?

Reply:
> The line below looks incorrect. May be it is supposed to be `if keepdims == 0 and split is None` ?

This test is checking that case: https://github.com/onnx/onnx/pull/4831/files#diff-e88ac7f22fe3c2c4e48cc6e90a49600312dde1f3e7757e0427426b0543ebfb87R2959.

Reply:
But why is the condition `axis is None` ? It doesn't depend on the axis attribute. It depends on whether the optional input split is specified, right?

Reply:
True, axis cannot be None aynway. I remove this condition.

Reply:
Thanks. But I think we do need the condition `and split is None`. Consider [this line of the spec](https://github.com/onnx/onnx/blob/main/onnx/defs/sequence/defs.cc#L299) ... in general, removing the dim does not make sense unless that dim has a value of 1 in each tensor in the sequence. So, the spec kind of makes sense, and it is better if we align with this exactly.

Reply:
Sorry, I missed it. Argument `keepdims` is pretty common. I probably never read this line. I did not expect the value to be ignored in some cases.